fieldservice_geoengine/tests/test_fsm_location.py

```diff
@@ -1,52 +1,154 @@
 
 
-class TestFsmLocation(TransactionCase):
-    def setUp(self):
-        super(TestFsmLocation, self).setUp()
-        self.fsm_location = self.env["fsm.location"]
-        self.location_partner_1 = self.env.ref("fieldservice.location_partner_1")
-        self.location_partner_2 = self.env.ref("fieldservice.location_partner_2")
-        self.test_loc_partner = self.env.ref("fieldservice.test_loc_partner")
-        self.location_partner_3 = self.env.ref("fieldservice.location_partner_3")
+class TestFsmLocation(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.FSMLocation = cls.env["fsm.location"]
+        cls.location_partner_1 = cls.env.ref("fieldservice.location_partner_1")
+        cls.location_partner_2 = cls.env.ref("fieldservice.location_partner_2")
+        cls.location_partner_3 = cls.env.ref("fieldservice.location_partner_3")
+        cls.test_loc_partner = cls.env.ref("fieldservice.test_loc_partner")
+        # delta value of 0.00001 was chosen according to OpenStreetMap's decimal precision table
+        # https://wiki.openstreetmap.org/wiki/Precision_of_coordinates#Conversion_to_decimal
+        # 5 decimals are necessary for precision of about a metre
+        cls.delta = 0.000_01
 
-    def test_fsm_location(self):
-        test_location = self.fsm_location.create(
+        cls.test_location = cls.FSMLocation.create(
             {
                 "name": "Test Location 2",
                 "phone": "123",
-                "email": "tl@email.com",
+                "email": "test@example.com",
+                "partner_id": cls.location_partner_1.id,
+                "owner_id": cls.location_partner_2.id,
+            }
+        )
+
+    def test_fsm_location_creation(self):
+        test_partner = self.env["res.partner"].create(
+            {
+                "name": "Test partner",
+            }
+        )
+        # should not be localized yet
+        test_partner.write(
+            {
+                "street": "Rue des Bourlottes 9",
+                "zip": "1367",
+                "city": "Grand-Rosière",
+                "country_id": self.env.ref("base.be"),
+            }
+        )
+        self.assertFalse(self.location_partner_1.partner_latitude)
+        self.assertFalse(self.location_partner_1.partner_longitude)
+        # should be localized after assigning a partner to the location
+        test_location_1 = self.FSMLocation.create(
+            {
+                "name": "Test Location 2",
+                "phone": "123",
+                "email": "test@example.com",
+                "partner_id": test_partner.id,
+                "owner_id": self.location_partner_1.id,
+            }
+        )
+        self.assertTrue(test_location_1.partner_latitude)
+        self.assertTrue(test_location_1.partner_longitude)
+        self.assertAlmostEqual(
+            test_location_1.partner_latitude, 50.629981, delta=self.delta
```

Comment:
now, tests are failling on 14.0 because the value is too precise
 
```
 2023-05-17 09:50:34,282 254 ERROR odoo odoo.addons.fieldservice_geoengine.tests.test_fsm_location: FAIL: TestFsmLocation.test_fsm_location_creation
Traceback (most recent call last):
  File "/__w/field-service/field-service/fieldservice_geoengine/tests/test_fsm_location.py", line 63, in test_fsm_location_creation
    test_location_1.partner_latitude, 50.629981, delta=self.delta
AssertionError: 50.629850000000005 != 50.629981 within 1e-05 delta
``` 
like https://github.com/OCA/field-service/actions/runs/5001740733/jobs/8960810974#step:8:167


both values looks legit 

@PicchiSeba  do you know why the results are not consistent  ?
can we reduce the delta to like 0,001 or may write a different test

Reply:
Is this an isolated test failure or has it happened before?
We can loosen the precision from 0.000_01 to 0.000_1 to temporarily fix the pipelines.

I think loosening it more that that would be bad to check the correct functionality.
That digit we are skipping (from 5 to 4) means check for precision of ~10 metres instead of ~1 meter.

Reply:
Each commit on 14.0 has this error (like translation updates).

>I think loosening it more than that would be bad to check the correct module functionality.

Yes but, is it the responsibility of this module to test the trueness of this adress lookup ?

Reply:
That's a good point actually.

This module should just integrate base_geoengine with fieldservice. It is not this module's responsibility to actually check the correctness of all the geolocation calculations

Reply:
fixed  in https://github.com/OCA/field-service/pull/1097